exchanges/kraken/kraken_test.go

```diff
@@ -44,33 +45,19 @@ const (
 	canManipulateRealOrders = false
 )
 
-// TestSetup setup func
 func TestMain(m *testing.M) {
-	k.SetDefaults()
-	cfg := config.GetConfig()
-	err := cfg.LoadConfig("../../testdata/configtest.json", true)
-	if err != nil {
-		log.Fatal(err)
-	}
-	krakenConfig, err := cfg.GetExchangeConfig("Kraken")
-	if err != nil {
-		log.Fatal(err)
-	}
-	krakenConfig.API.AuthenticatedSupport = true
-	krakenConfig.API.Credentials.Key = apiKey
-	krakenConfig.API.Credentials.Secret = apiSecret
-	k.Websocket = sharedtestvalues.NewTestWebsocket()
-	err = k.Setup(krakenConfig)
-	if err != nil {
-		log.Fatal(err)
-	}
-	err = k.UpdateTradablePairs(context.Background(), true)
-	if err != nil {
+	k = new(Kraken)
+	if err := testexch.TestInstance(k); err != nil {
```

Comment:
Doing this and leaving empty api key strings above will need to be amended.
This seems related to this snippet:
> One thing I'm unhappy about is the change of Key/Secret for this to work.
I really don't like that there's special values for the auth values which get silently ignored, such as Test, Mock, Key, Secret, etc.
So I've ended up using something adhoc, and putting a require in the test to ensure it's obvious if the auth-conn isn't connected.

But using the old testmain, your `TestWsCancelOrders` test still functions. I'm a bit unsure specifically what you're referring to in that snippet anyway

Reply:
Thanks. Nope, this was just a mistake, the apiKeys should have been respected.
( _It would be passive-aggressive-AF to yank this mechanism without discussing further with team members who have said it's their current methodology_ )

Fixed 4f95ad49